packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -2136,7 +2136,7 @@ class _InputDecoratorState extends State<InputDecorator> with TickerProviderStat
       opacity: (isEmpty && !_hasInlineLabel) ? 1.0 : 0.0,
       duration: _kTransitionDuration,
       curve: _kTransitionCurve,
-      alwaysIncludeSemantics: true,
+      alwaysIncludeSemantics: isEmpty || (decoration.labelText == null && decoration.label == null),
```

Comment:
I think remove this line will be good enough (defaults to false). Since it will automatically remove the semantics when opacity is 0.0

Reply:
What about the case that there is no label and there is text in the textfield.

As discussed in #114511, in that case the hint is likely the label do we do want to include the semantics although it's not visible.

Reply:
I am not sure we would still want to include the semantics if there is no label and and there is text in the textfield. In this case the hinttext will still be withdrawn, and it seemed weird to announce something that is not on the screen. I know this is conflicting with what @goderbauer was suggesting. We should have a discussion if we cannot resolve the conversation here

Reply:
I think we should still keep the hint if there is no label with the reason I gave before. It's not uncommon to announce something that's not drawn on screen (e.g. we also announce the tooltip as label for some button even when it is not drawn on screen).

Reply:
yeah, I guess that can be possible use case. I am ok with this as is.